src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -691,6 +691,7 @@ public sealed override Type[] GetForwardedTypes()
             return types.ToArray();
         }
 
+        [RequiresUnreferencedCode("Types might be removed")]
```

Comment:
I think it's worth adding a comment that the problem is recursivenes. This method returns all nested types recursively, which `DynamicallyAccessedMember` can't express.